kyo-caliban/src/test/scala/kyo/resolversTest.scala

```diff
@@ -0,0 +1,134 @@
+package kyo
+
+import caliban.*
+import caliban.render
+import caliban.schema.Schema
+import caliban.schema.SchemaDerivation
+import kyo.*
+import kyoTest.KyoTest
+import sttp.model.Uri
+import sttp.tapir.*
+import sttp.tapir.json.zio.*
+import zio.Task
+import zio.ZLayer
+
+class resolversTest extends KyoTest:
+
+    def runZIO[T](v: Task[T]): T =
+        zio.Unsafe.unsafe(implicit u =>
+            zio.Runtime.default.unsafe.run(v).getOrThrow()
+        )
+
+    case class Query(
+        k1: Int < Aborts[Throwable],
+        k2: Int < ZIOs,
+        k3: Int < (Aborts[Throwable] & ZIOs),
+        k4: Int < IOs
+    ) derives Schema.SemiAuto
+
+    "schema derivation" in {
+        val expected = """type Query {
+                         |  k1: Int
+                         |  k2: Int
+                         |  k3: Int
+                         |  k4: Int
+                         |}""".stripMargin
+        assert(render[Query].trim == expected)
+    }
+
+    "execution" in runZIO {
+        val api = graphQL(RootResolver(Query(42, 42, 42, 42)))
+        for
+            interpreter <- api.interpreter
+            res         <- interpreter.execute("{ k1 k2 k3 k4 }")
+        yield assert(res.data.toString == """{"k1":42,"k2":42,"k3":42,"k4":42}""")
+        end for
+    }
+
+    "arbitrary kyo effects" in runZIO {
+        type Env = Vars[Int] & Consoles
+        object schema extends SchemaDerivation[Runner[Env]]
+
+        case class Query(k: Int < Env) derives schema.SemiAuto
+
+        val api = graphQL(RootResolver(Query(
+            for
+                _ <- Vars.update[Int](_ + 1)
+                v <- Vars.get[Int]
+                _ <- Consoles.println(v)
```

Comment:
I'd prefer to avoid printing to the console in tests. Maybe showing `Envs` use would be interesting here instead?

Reply:
Sure, I picked that one arbitrarily. Changed to `Envs`.